examples/tree_3d_dgsem/elixir_mhd_dynamo.jl

```diff
@@ -0,0 +1,167 @@
+# A turbulent helically driven dynamo model in resistive compressible MHD.
+# We start with a weak small-scale magnetic field provided as Gaussian noise.
+# The energy is supplied via a forcing function f_force(x, t), which is delta-correlated in time.
+# For a comparison look into the PC run pencil-code/simon/forced/alpha2_periodic/64_kf_5_a.
+
+using OrdinaryDiffEq
+using Trixi
+using Random
+using LinearAlgebra
+
+###############################################################################
+# initial condition and forcing function
+
+"""
+    initial_condition_gaussian_noise_mhd(x, t, equations)
+
+Define the initial condition with a weak small-scale Gaussian noise magnetic field.
+"""
+function initial_condition_gaussian_noise_mhd(x, t, equations)
+    amplitude = 1e-8
+
+    seed = reinterpret(UInt64, t + x[1] + x[2] + x[3])
+    Random.seed!(seed)
+
+    rho = 1.0
+    rho_v1 = 0.0
+    rho_v2 = 0.0
+    rho_v3 = 0.0
+    rho_e = 10.0
+    B1 = randn(1)[1] * amplitude
+    B2 = randn(1)[1] * amplitude
+    B3 = randn(1)[1] * amplitude
```

Comment:
Do you mean that for the same seeds we could get different rando numbers using different Julia versions? That would be an issue for the test.  Do we allow distinctions in the tests with regard to the Julia version? That way we could just add an `if` condition.

Reply:
Yes, that's what I mean. I would detest a statement like `if VERSION ...` in code like this. You may look at https://github.com/JuliaRandom/StableRNGs.jl for something that is less efficient but stable. But I would still like to know whether there is another way to initialize this setup and get a review from other core devs, e.g., @andrewwinters5000

Reply:
`StableRNG` looks promising. However, they might not be entirely free from update issues:
_The_ StableRNG type provided by this package strives for stability, but if bugs which require breaking this promise are found, a new major version will be released with the fix._
Considering their library consists of a single file, maybe we can include it as third party software in Trixi.jl. That way we would have control over the version. But that is a serious design question.

Reply:
I wouldn't include it. They basically say that that only change the stream if it's crap due to bugs. In this case, we definitely want to update our tests anyway.

Reply:
And they say it will be a new major version, so it will not hit us unexpected but can be controlled by `test/Project.toml` compatibility settings.

Reply:
Seems like StableRNGs is a viable solution. Since we specify the version number we should not have any unexpected test breakage. Just pushed my changes.